### PR TITLE
Add Statuses to NOIs

### PR DESCRIPTION
**Pull request by dhaselhan**

### CODE REVIEW

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.controller.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { UserRoles } from '../../../common/authorization/roles.decorator';
+import { NoticeOfIntentSubmissionToSubmissionStatusDto } from './notice-of-intent-status.dto';
+import { NoticeOfIntentSubmissionToSubmissionStatus } from './notice-of-intent-status.entity';
+import { NoticeOfIntentSubmissionStatusService } from './notice-of-intent-submission-status.service';
+
+@Controller('notice-of-intent-submission-status')
+@UserRoles(...ANY_AUTH_ROLE)
+export class NoticeOfIntentSubmissionStatusController {
+  constructor(
+    private applicationSubmissionStatusService: NoticeOfIntentSubmissionStatusService,
```
Comment (sandratoh) [inline services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.controller.ts:14]: Rename `applicationSubmissionStatusService` to something noi-related

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+@Injectable()
+export class NoticeOfIntentSubmissionStatusService {
+  constructor(
+    @InjectRepository(NoticeOfIntentSubmissionToSubmissionStatus)
+    private statusesRepository: Repository<NoticeOfIntentSubmissionToSubmissionStatus>,
+    @InjectRepository(NoticeOfIntentSubmissionStatusType)
+    private submissionStatusTypeRepository: Repository<NoticeOfIntentSubmissionStatusType>,
+    @InjectRepository(NoticeOfIntentSubmission)
+    private applicationSubmissionRepository: Repository<NoticeOfIntentSubmission>,
```
Comment (sandratoh) [inline services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts:24]: Rename to `noticeOfIntentSubmissionRepository` or something without application

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  private async getSubmission(fileNumber: string) {
+    const submission = await this.applicationSubmissionRepository.findOneBy({
+      fileNumber,
+    });
+
+    if (!submission) {
+      throw new ServiceNotFoundException(
+        `Submission does not exist for provided application ${fileNumber}. Only applications originated in portal have statuses.`,
```
Comment (sandratoh) [inline services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts:112]:
```suggestion
        `Submission does not exist for provided notice of intent ${fileNumber}. Only notices of intent originated in portal have statuses.`,
```

**Review by sandratoh — APPROVED** [review]
Just need to rename a few things but LGTM 👍

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.controller.ts (hunk 1 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.controller.ts]: Done!

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts (hunk 2 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts]: Done!

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts (hunk 3 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts]: Done!

**Pull request merged**